packages/providers/src/transaction-request/transaction-request.ts

```diff
@@ -255,12 +260,12 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
    * @param predicate - Predicate bytes.
    * @param predicateData - Predicate data bytes.
    */
-  addCoinInput(coin: Coin, predicate?: AbstractPredicate) {
+  addCoinInput<A extends AbstractAccount>(coin: AccountResource<Coin, A>) {
     const { assetId, owner, amount } = coin;
 
     let witnessIndex;
 
-    if (predicate) {
+    if ((coin.account as AbstractPredicate).bytes) {
```

Comment:
It's not possible to use `account instanceof AbstractPredicate` because `Predicate` itself just implements `AbstractPredicate`, it doesn't extend it; it extends `Account`.